src/domain/settings/regions/details.tsx

```diff
@@ -0,0 +1,343 @@
+  useAdminRegion,
+  useAdminRegionFulfillmentOptions,
+  useAdminStore,
+  useAdminStorePaymentProviders,
+  useAdminUpdateRegion,
+} from "medusa-react"
+import React, { useEffect, useState } from "react"
+import { useForm } from "react-hook-form"
+import { Box, Flex } from "rebass"
+import DuplicateIcon from "../../../components/fundamentals/icons/duplicate-icon"
+import TrashIcon from "../../../components/fundamentals/icons/trash-icon"
+import Input from "../../../components/molecules/input"
+import Select from "../../../components/molecules/select"
+import BodyCard from "../../../components/organisms/body-card"
+import DeletePrompt from "../../../components/organisms/delete-prompt"
+import Spinner from "../../../components/spinner"
+import useToaster from "../../../hooks/use-toaster"
+import { countries as countryData } from "../../../utils/countries"
+import { getErrorMessage } from "../../../utils/error-messages"
+import fulfillmentProvidersMapper from "../../../utils/fulfillment-providers.mapper"
+import paymentProvidersMapper from "../../../utils/payment-providers-mapper"
+import Shipping from "./shipping"
+
+const RegionDetails = ({ id, onDelete, handleSelect }) => {
+  const [currencies, setCurrencies] = useState([])
+  const [countries, setCountries] = useState([])
+  const [selectedCurrency, setSelectedCurrency] = useState(null)
+  const [paymentOptions, setPaymentOptions] = useState([])
+  const [paymentProviders, setPaymentProviders] = useState([])
+  const [fulfillmentOptions, setFulfillmentOptions] = useState([])
+  const [fulfillmentProviders, setFulfillmentProviders] = useState([])
+
+  const { register, reset, setValue, handleSubmit } = useForm()
+  const toaster = useToaster()
+
+  const { store, isLoading: storeIsLoading } = useAdminStore()
+  const createRegion = useAdminCreateRegion()
+  const deleteRegion = useAdminDeleteRegion(id)
+  const { region, isLoading: regionIsLoading } = useAdminRegion(id)
+  const {
+    payment_providers,
+    isLoading: paymentIsLoading,
+  } = useAdminStorePaymentProviders()
+  const {
+    fulfillment_options,
+    isLoading: fulfillmentIsLoading,
+  } = useAdminRegionFulfillmentOptions(id)
+  const updateRegion = useAdminUpdateRegion(id)
+
+  const [showDanger, setShowDanger] = useState(false)
+
+  useEffect(() => {
+    if (!store || !region) return
+    register({ name: "currency_code" })
+    setValue("currency_code", region.currency_code)
+    setSelectedCurrency({
+      value: region.currency_code,
+      label: region.currency_code.toUpperCase(),
+    })
+    setCurrencies(getCurrencies(store.currencies))
+  }, [store, region])
+
+  useEffect(() => {
+    if (paymentIsLoading) return
+    setPaymentOptions(payment_providers.map(c => paymentProvidersMapper(c.id)))
+  }, [payment_providers, paymentIsLoading])
+
+  useEffect(() => {
+    if (fulfillmentIsLoading) return
+    setFulfillmentOptions(
+      fulfillment_options.map(c => fulfillmentProvidersMapper(c.provider_id))
```

Comment:
this throws when I run it locally, probably because i'm using the tekla db so there might be an isssue with that. However, is `useAdminFulfillmentOptions` fails for whatever reason, this will throw because `fulfillment_options` will be undefined, so maybe add a tiny `?`. 

Note that if you use `?`, `fulfillmentOptions` (the local state) will be undefined so maybe default to an empty array or use a plain old boring `if (fulfillment_options) setFulfillmentOptions(...)`